test/functional/api/vim_spec.lua

```diff
@@ -1267,5 +1268,44 @@ describe('API', function()
       eq(expected, nvim("list_uis"))
     end)
   end)
-
+  describe('statusline', function()
+    it('last char shows (multibyte)', function()
+      local screen = Screen.new(20, 4)
+      screen:attach()
+      command('set laststatus=2')
+      command('set statusline=你好')
+      screen:expect([[
+      ^                    |
+      ~                   |
+      你好                |
+                          |
+      ]])
+    end)
+  end)
+  describe('statusline', function()
+    it('last char shows', function()
+      local screen = Screen.new(20, 4)
+      screen:attach()
+      command('set laststatus=2')
+      command('set statusline=abc')
+      screen:expect([[
+      ^                    |
+      ~                   |
+      abc                 |
+                          |
+      ]])
+    end)
+    it('unicode control points', function()
+      local screen = Screen.new(20, 4)
+      screen:attach()
+      command('set laststatus=2')
+      feed(':set statusline=<C-v>u9f<SPACE><CR>')
+      screen:expect([[
+      ^                    |
+      ~                   |
+      <9f>                |
+                          |
+      ]])
+    end)
+  end)
```

Comment:
We could add a test with up to `MAX_MCO` composing characters.

Reply:
Should this test work? it only renders one <9f> in the bottom at the moment
```    it('MAX_MCO unicode control points', function()                                  
      local screen = Screen.new(24, 4)                                               
      screen:attach()                                                                
      command('set laststatus=2')                                                    
      feed([[                                                                        
      :set statusline=<C-v>u9f                                                       
      <C-v>u9f                                                                       
      <C-v>u9f                                                                       
      <C-v>u9f                                                                       
      <C-v>u9f                                                                       
      <C-v>u9f<SPACE><CR>                                                            
      ]])                                                                            
      screen:expect([[                                                               
      ^                        |                                                     
      ~                       |                                                      
      <9f><9f><9f><9f><9f><9f>|                                                      
                              |                                                      
      ]])                                                                            
    end)                                                                             
```

Reply:
Try putting all the character on one line: `:set statusline=<C-v>u9f<C-v>u9f<C-v>u9f<C-v>u9f<C-v>u9f<C-v>u9f<CR>`

Also `U+009f` is not a [composing character](https://en.wikipedia.org/wiki/Combining_character). To input unicode with more than two hex digits you can use `<C-v>U...`:
a + U+fe20: a︠
a + U+fe20 + U+fe21: a︠︡

Reply:
This behaves weird: 

```
    it('MAX_MCO (6) unicode control points', function()                              
      local screen = Screen.new(24, 4)                                               
      screen:attach()                                                                
      command('set laststatus=2')                                                    
      feed(string.format(":set statusline=a%s%s%s%s%s%s<CR>",                          
      "<C-v>ufe20", "<C-v>ufe21", "<C-v>u20df", "<C-v>ufe26",                          
      "<C-v>ufe2e", "<C-v>u20de"))                                                     
      screen:expect([[                                                                 
      ^                        |                                                       
      ~                       |                                                        
      a︠︡⃟︦︮⃞                       |                                                        
                              |                                                        
      ]])                                                                              
    end)
```

results in 
```
Expected:
  |^                        |
  |~                       |
  |a︠︡⃟︦︮⃞                       |
  |*                        |
Actual:
  |^                        |
  |~                       |
  |a︠︡⃟︦︮⃞                       |
  |*:set statusline=a︠︡⃟︦︮⃞       |
```
although the same command works without composing characters. I'm not sure why that does not work ( the 6 composing chars are rendered correctly )